sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -371,9 +371,12 @@ RT::PiProgram ProgramManager::getBuiltPIProgram(OSModuleHandle M,
     // If device image is not SPIR-V, DeviceLibReqMask will be 0 which means
     // no fallback device library will be linked.
     uint32_t DeviceLibReqMask = 0;
-    if (Img.getFormat() == PI_DEVICE_BINARY_TYPE_SPIRV &&
+    // FIXME: disable the fallback device libraries online link as not all
+    // backend supports spv online link. Need to enable it when all backends
+    // support spv online link.
+    /* if (Img.getFormat() == PI_DEVICE_BINARY_TYPE_SPIRV &&
         !SYCLConfig<SYCL_DEVICELIB_NO_FALLBACK>::get())
-      DeviceLibReqMask = getDeviceLibReqMask(Img);
+      DeviceLibReqMask = getDeviceLibReqMask(Img); */
```

Comment:
There is already code elsewhere in this file to disable online linking of device libraries for the Level Zero backend.  Rather than introduce a new place that disables online linking, I think it might be better to change the existing code that does this for Level Zero.

I suggest changing this code in `ProgramManager::build()`:

```
  // The Level Zero driver support for online linking currently has bugs, but
  // we think the DPC++ runtime support is ready.  This environment variable
  // gates the runtime support for online linking, so we can try enabling if a
  // new driver is released before the next DPC++ release.
  static bool EnableLevelZeroLink = std::getenv("SYCL_ENABLE_LEVEL_ZERO_LINK");
  if (!EnableLevelZeroLink) {
    if (Context->getPlugin().getBackend() == backend::level_zero) {
      LinkDeviceLibs = false;
    }
  }
```

to something like:

```
  // TODO: Because online linking isn't implemented yet on Level Zero, the
  // compiler always links against the fallback device libraries.  Once
  // online linking is supported on all backends, we should remove the line
  // below and also change the compiler, so it no longer links the fallback
  // code unconditionally.
  LinkDeviceLibs = false;
```

Reply:
Hi, @gmlueck 
Already fixed. Thank you very much.